### PR TITLE
Task/api 324 dr read and sentinel

### PR DESCRIPTION
**Pull request by ilaflamme-va**
Read and Sentinel: https://vasdvp.atlassian.net/browse/API-324

A decision needs to me made regarding the Category - Coding - System field.
Per the spec, https://www.fhir.org/guides/argonaut/r2/StructureDefinition-argo-diagnosticreport.html, 

  Required Pattern: {"coding":[{"system":"http://hl7.org/fhir/ValueSet/diagnostic-service- 
  sections","code":"LAB"}]}

Currently the sandbox database is returning 

 "category": {
        "coding": [
            {
                "system": "http://hl7.org/fhir/v2/0074",
                "code": "CH",
                "display": "Chemistry"
            },

causing validation to fail.

### CODE REVIEW

File: argonaut-api/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE suppressions PUBLIC
+  "-//Checkstyle//DTD SuppressionFilter Configuration 1.2//EN"
+  "https://checkstyle.org/dtds/suppressions_1_2.dtd">
+<suppressions>
+  <!--
+    Diagnostic report _status and _issued are data absent reason fields.
+  -->
+  <suppress checks="MemberName" files=".*/DiagnosticReport.java" message="Member name '(_status|_issued)'"/>
```
Comment (bschofield-va) [inline argonaut-api/checkstyle-suppressions.xml]: @esubocz-va @ilaflamme-va @eclendenning-va 
Be aware to support FHIR data absent reason fields, I had to introduce checkstyle suppressions for this project.

File: argonaut-api/src/test/java/gov/va/api/health/argonaut/api/AbstractRelatedFieldVerifier.java [hunk 2]
```diff
@@ -26,25 +27,23 @@
...
 import org.springframework.beans.BeanUtils;
 
-/**
- * This class will verify fields with a given prefix are properly configured in the same ZeroOrOneOf
- * group. This class will fiend related fields with the same prefix and systematically test
- * different combinations to ensure they are validate as expected.
- */
 @Slf4j
-public class ZeroOrOneVerifier<T> {
+public abstract class AbstractRelatedFieldVerifier<T> {
```
Comment (bschofield-va) [inline argonaut-api/src/test/java/gov/va/api/health/argonaut/api/AbstractRelatedFieldVerifier.java]: @eclendenning-va I tore your ZeroOrOneOf verifier up ... splitting it into some good reusable parts for the new exactly one of validations.

**Comment by bschofield-va** [comment]
This is ready to go pending an update to the CDW schemas samples, currently blocked by db team. Once this is ready, we'll update the cdw-schemas version and merge.

**Comment by ilaflamme-va** [comment]
Consolidating Read and Search.

**Pull request closed without merging**